packages/image_picker/image_picker_ios/test/image_picker_ios_test.dart

```diff
@@ -0,0 +1,937 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:image_picker_ios/image_picker_ios.dart';
+import 'package:image_picker_ios/src/messages.g.dart';
+import 'package:image_picker_platform_interface/image_picker_platform_interface.dart';
+
+import 'test_api.dart';
+
+@immutable
+class _LoggedMethodCall {
+  const _LoggedMethodCall(this.name, {required this.arguments});
+  final String name;
+  final Map<String, Object?> arguments;
+
+  @override
+  bool operator ==(Object other) {
+    return other is _LoggedMethodCall &&
+        name == other.name &&
+        mapEquals(arguments, other.arguments);
+  }
```

Comment:
tangent: Dart equality is kind of a mess.  You never know if `==` means identity or value equality.  I know there is prior art for mixing these up.